queries.sql

```diff
@@ -93,4 +93,59 @@ SELECT owners.full_name, COUNT(animals)
 FROM animals
 JOIN owners ON owners.id = animals.owner_id
 GROUP BY owners.full_name
-ORDER BY COUNT(animals) DESC LIMIT 1;
\ No newline at end of file
+ORDER BY COUNT(animals) DESC LIMIT 1;
+
+
+SELECT COUNT(animals)
+FROM animals
+JOIN visits ON visits.animal_id = animals.id
+JOIN vets ON vets.id = visits.vet_id
+WHERE vets.name = 'Stephanie Mendez';
+
+SELECT vets.name, species.name
+FROM vets
+FULL JOIN specializations ON specializations.vet_id = vets.id
+FULL JOIN species ON species.id = specializations.species_id;
+
+SELECT animals.name, visits.visit_date
+FROM animals
+JOIN visits ON visits.animal_id = animals.id
+JOIN vets ON vets.id = visits.vet_id
+WHERE vets.name = 'Stephanie Mendez'
+AND visits.visit_date BETWEEN '2020-04-01' AND '2020-08-30';
+
+SELECT animals.name, COUNT(visits)
+FROM animals
+join visits on visits.animal_id = animals.id
+GROUP BY animals.name
+ORDER BY COUNT(visits) DESC LIMIT 1;
+
+SELECT animals.name, visits.visit_date
+FROM animals
+JOIN visits ON visits.animal_id = animals.id
+JOIN vets ON vets.id = visits.vet_id
+WHERE vets.name = 'Maisy Smith'
+ORDER BY visits.visit_date LIMIT 1;
+
+SELECT animals.name, vets.name, visits.visit_date
+FROM animals
+JOIN visits ON visits.animal_id = animals.id
+JOIN vets ON vets.id = visits.vet_id
+ORDER BY visits.visit_date DESC LIMIT 1;
+
+SELECT vets.name, COUNT (visits)
+from vets
+FULL JOIN Specializations ON Specializations.vet_id = vets.id
+FULL JOIN species ON species.id = specializations.species_id
+JOIN visits ON visits.vet_id = vets.id
+WHERE species.name IS NULL
```

Comment:
- [ ]  **[OPTIONAL]** Instead of using `NULL` operator I think it would be better if you use the `NOT IN` operator. This will help to avoid SQLexceptions in some cases. 
- Example here:
```
  SELECT count(*)
  FROM visits
  LEFT JOIN animals ON animals.id = visits.animal_id
  LEFT JOIN vets ON vets.id = visits.vet_id
  WHERE animals.species_id NOT IN (SELECT species_id FROM specializations WHERE vet_id = vets.id)

```